libres/lib/enkf/enkf_main.cpp

```diff
@@ -743,6 +737,337 @@ static FILE *enkf_main_log_step_list(const char *log_path,
     return log_stream;
 }
 
+static std::unordered_map<std::string, matrix_type *>
+enkf_main_load_parameters_from_ministep(
+    enkf_fs_type *target_fs, ensemble_config_type *ensemble_config,
+    int_vector_type *iens_active_index, int last_step, run_mode_type run_mode,
+    meas_data_type *forecast, enkf_state_type **ensemble, hash_type *use_count,
+    obs_data_type *obs_data, const local_ministep_type *ministep) {
+
+    int cpu_threads = 4;
+    thread_pool_type *tp = thread_pool_alloc(cpu_threads, false);
+    int matrix_start_size = 250000;
+    int active_ens_size = meas_data_get_active_ens_size(forecast);
+    matrix_type *A = matrix_alloc(matrix_start_size, active_ens_size);
+
+    serialize_info_type *serialize_info = serialize_info_alloc(
+        target_fs, //src_fs - we have already copied the parameters from the src_fs to the target_fs
+        target_fs, ensemble_config, iens_active_index, 0, ensemble, run_mode,
+        last_step, A, cpu_threads);
+
+    std::unordered_map<std::string, matrix_type *> parameters;
+    for (auto &[dataset_name, dataset] :
+         local_ministep_get_datasets(ministep)) {
+
+        if (!(local_dataset_get_size(dataset)))
+            continue;
+
+        local_obsdata_type *local_obsdata =
+            local_ministep_get_obsdata(ministep);
+        const auto &unscaled_keys = local_dataset_unscaled_keys(dataset);
+
+        if (unscaled_keys.size() == 0)
+            continue;
+
+        // Part 1: Parameters which do not have row scaling attached.
+        enkf_main_serialize_dataset(ensemble_config, dataset, last_step,
```

Comment:
I find this a bit hard to read.
From what I understand, `serialize_info` contains a pointer to `A` which gets updated in `enkf_main_serialize_dataset`.
`A` is then copied to `A_new`.

What if we instead send `A` directly yo `enkf_main_serialize_dataset`, or set `A_new` as:

`matrix_type *A_new = matrix_alloc_copy(serialize_info->A)`?